src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -1100,7 +1100,15 @@ private void ProcessDownloadQueueLocked()
                     return;
                 }
 
+                // To fix issue https://github.com/stratisproject/StratisBitcoinFullNode/issues/2294#issue-364513736
+                // if there are no samples, assume the worst scenario (you are going to donwload full blocks).
                 long avgSize = (long)this.blockPuller.GetAverageBlockSizeBytes();
+                if (avgSize == 0)
+                {
+                    // TODO: move MaxBlockSize on a common place.
+                    avgSize = Block.MaxBlockSize;
```

Comment:
No this is wrong it should come from `Netowrk.Consensus`

Reply:
Yes. We have this value copypasted all over. Should just refactor the code to make sure it's in one place. 

Probably next thing @MithrilMan can do is to find all block size values and making sure we use only one from Network.Consensus

Reply:
That's why there is a TODO. 
I think @MithrilMan already discussed that in fn channel

Reply:
yes, sparse delcarations all around
I can try to replace all Block.MaxBlockSize to `network.Consensus.Options.MaxBlockBaseSize` if that's the current property (what does `Base` refers to?)

Reply:
yes @MithrilMan please replace to `network.Consensus.Options.MaxBlockBaseSize` and get rid of all other places with a block size.

Reply:
actually there are 16 references, would be quick to just update last one and then looking to change them all (not just Block. but even other const around, like Transaction.MAX_BLOCK_SIZE)

Reply:
#2375